compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/RootNodeOwner.skiko.kt

```diff
@@ -396,11 +398,15 @@ internal class RootNodeOwner(
 
         override val focusOwner: FocusOwner = FocusOwnerImpl(platformFocusOwner, this)
 
-        val rootModifier = if (ComposeUiFlags.areWindowInsetsRulersEnabled) {
-                RulerProviderModifierElement(platformContext.windowInsets)
-            } else {
-                Modifier
-            }
+        val rootModifier = Modifier
+            .then(RootWindowInsetsProviderModifierElement(platformContext.windowInsets))
+            .then(
+                if (ComposeUiFlags.areWindowInsetsRulersEnabled) {
+                    RulerProviderModifierElement(platformContext.windowInsets)
+                } else {
+                    Modifier
+                }
+            )
```

Comment:
Nitpick: a helper function might help with readability and avoiding adding extra empty `Modifier`

```kt
private fun Modifier.rulerProvider(windowInsets: PlatformWindowInsets) =
    if (ComposeUiFlags.areWindowInsetsRulersEnabled) then(RulerProviderModifierElement(windowInsets)) else this
```

Reply:
I like that, added